gordian/repo.py

```diff
@@ -104,15 +109,32 @@ def find_file(self, filename):
             if file.path == filename:
                 return file
 
-    def make_branch(self):
-        sb = self._repo.get_branch(self.target_branch)
+    def _make_branch(self):
+        logger.info('Forking repo...')
+        self._forked_repo = self._original_repo.create_fork()
+        branch = self._get_branch()
+        logger.debug(f'Creating branch {self.branch_name}')
+
         try:
-            logger.debug(f'Creating branch {self.branch_name}')
-            ref = self._repo.create_git_ref(ref=self.branch_name, sha=sb.commit.sha)
+            ref = self._forked_repo.create_git_ref(ref=self.branch_name, sha=branch.commit.sha)
         except GithubException as e:
-            print(f"Branch {self.branch_name} already exists in github")
+            logger.debug(f'Branch {self.branch_name} already exists in github')
         self.branch_exists = True
 
+    def _get_branch(self, tries=1):
```

Comment:
would you consider to use https://pypi.org/project/retry/
```python
@retry(GithubException, tries=3, delay=1, backoff=2)
def _get_branch(self):
 ...
```

Reply:
Mind blown, did not know about this... 🤯